ui/app/components/app/metamask-template-renderer/metamask-template-renderer.js

```diff
@@ -0,0 +1,102 @@
+import React, { memo } from 'react'
+import PropTypes from 'prop-types'
+import { isEqual } from 'lodash'
+import { safeComponentList } from './safe-component-list'
+
+function getElement(section) {
+  const { element } = section
+  const Element = safeComponentList[element]
+  if (!Element) {
+    throw new Error(
+      `${element} is not in the safe component list for MetaMask template renderer`,
+    )
+  }
+  return Element
+}
+
+const MetaMaskTemplateRenderer = ({ sections }) => {
+  if (!sections) {
+    // If sections is null eject early by returning null
+    return null
+  } else if (typeof sections === 'string') {
+    // React can render strings directly, so return the string
+    return sections
+  } else if (
+    sections &&
+    typeof sections === 'object' &&
+    !Array.isArray(sections)
+  ) {
+    // If dealing with a single entry, then render a single object without key
+    const Element = getElement(sections)
+    return (
+      <Element {...sections.props}>
+        {typeof sections.children === 'object' ? (
+          <MetaMaskTemplateRenderer sections={sections.children} />
+        ) : (
+          sections?.children
+        )}
```

Comment:
Could this be simplified? :thinking: 

```suggestion
        <MetaMaskTemplateRenderer sections={sections.children} />
```

Reply:
The same simplification could be made to the array case as well. Assuming it's valid here that is.

Reply:
Oh I see - maybe this was intentional, as an optimization, to avoid unnecessarily rendering a `MetaMaskTemplateRenderer` instance.

Reply:
it could, but then in the case of a string we're rendering an additional MetaMask template Renderer Render and it shows up in the tree of components in react profiler. I made this optimization to get keys onto the real components instead of onto the TemplateRenderer that rendered them. Example, if i commit your suggestion the keys for the buttons (ok-button, cancel-button) will be applied to intermediary `MetaMaskTemplateRenderers`

Reply:
Makes sense!